morf-core/src/main/java/org/alfasoftware/morf/sql/ParallelQueryHint.java

```diff
@@ -1,27 +1,65 @@
 package org.alfasoftware.morf.sql;
 
-import java.util.Objects;
+import static java.lang.String.format;
+
+import java.util.Optional;
+
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.apache.commons.lang3.builder.HashCodeBuilder;
+
 /**
  * Represents the hint for a parallel query.
  *
  * @author Copyright (c) Alfa Financial Software 2018
  */
 public final class ParallelQueryHint implements Hint {
 
+  private final Optional<Integer> degreeOfParallelism;
+
+  public ParallelQueryHint(Optional<Integer> degreeOfParallelism) {
+    this.degreeOfParallelism = degreeOfParallelism;
+  }
+
+
+  public ParallelQueryHint() {
+    this(Optional.empty());
+  }
+
+
+  /**
+   * @return the degree of parallelism for this PARALLEL query hint.
+   */
+  public Optional<Integer> getDegreeOfParallelism() {

```

Comment:
The return type here can remain Optional as it indicates to the caller that the value may not be present.